listenbrainz_spark/recommendations/tests/test_dataframe.py

```diff
@@ -0,0 +1,154 @@
+import uuid
+import unittest
+from datetime import datetime
+
+import listenbrainz_spark
+from listenbrainz_spark.tests import SparkTestCase
+from listenbrainz_spark.recommendations import create_dataframes
+from listenbrainz_spark import schema, utils, config, path, hdfs_connection, stats
+
+from pyspark.sql import Row
+
+# path used in between test functions of this class
+LISTENS_PATH = '/test/listens/' # HDFS path to listens
+MAPPING_PATH = '/test/mapping.parquet'
+MAPPED_LISTENS_PATH = '/test/mapped_listens.parquet'
+
+class CreateDataframeTestCase(SparkTestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.date = datetime.utcnow()
```

Comment:
Not trying to be nitpicky but this still is a class variable and not a instance variable. At this point, I think we've spent too much time on this anyways, so let's leave it be but just pointing this out.

https://dev.to/ogwurujohnson/distinguishing-instance-variables-from-class-variables-in-python-81

Reply:
I know the difference between instance var and class var (in theory) , I am just not sure how to do a self.date in a class method. Maybe you can take over it and I will surely learn a lot from you on how to put this is in practice. 

But thank you for so many informative links :)